### DISCUSSION TITLE
Is cloud storage emulator available?

### DISCUSSION THREAD

**Discussion by ithustle** [invertase/react-native-firebase#5330] [General]
I'd like to know if is already possible to use local cloud storage in an emulator?

**Comment by mikehardy** [comment]
Almost #5324

**Reply by mikehardy** [reply]
Firebase Storage Emulator support is live now in v12
There are still a few not-quite-right parts of it - you can see the tests in the storage e2e directory, mostly StorageReference marked FIXME where there are little emulator differences, but they'll get ironed out quickly I'm sure.

**Reply by ithustle** [reply]
Nice one. Thanks to keep me in post. I'll try it now

**Reply by ithustle** [reply]
@mikehardy , Has Firebase Storage Emulator integration with Firebase-Cli? I'm wondering which information I've to put here `firebase.storage().useEmulator({ host: ???, port: ??? })`? 

According with imagem below, Firebase Emulator Pack has not storage available

<img width="700" alt="Captura de ecrã 2021-05-19, às 12 49 31" src="https://user-images.githubusercontent.com/5114680/118807919-cc7db280-b8a0-11eb-97de-5027c35225fe.png">

**Reply by mikehardy** [reply]
You need firebase-tools 9.11.0 I believe. You can see how everything works in our tests app + CI configuration:

https://github.com/invertase/react-native-firebase/tree/master/.github/workflows/scripts
https://github.com/invertase/react-native-firebase/blob/fe3d22f54d0d706529eb5367795b568d9e783a4e/tests/package.json#L34

https://github.com/invertase/react-native-firebase/blob/fe3d22f54d0d706529eb5367795b568d9e783a4e/tests/app.js#L47

**Reply by ithustle** [reply]
Thanks

**Reply by ithustle** [reply]
That's weird...  I've updated today to 9.11.0 but storage emulator isn't available. I've put `storage: { 'port': 9199, host: '0.0.0.0' }`,  on `firebase.json` and when I tried to run I had this:

````
yarn run v1.22.10
$ firebase emulators:start
i  emulators: Starting emulators: auth, functions, database, storage
⚠  functions: The following emulators are not running, calls to these services from the Functions emulator will affect production: firestore, hosting, pubsub
✔  functions: Using node@16 from host.
i  database: Database Emulator logging to database-debug.log
i  emulators: Shutting down emulators.
i  functions: Stopping Functions Emulator
i  database: Stopping Database Emulator
i  auth: Stopping Authentication Emulator
i  hub: Stopping emulator hub

Error: Cannot start the Storage emulator without rules file specified in firebase.json: run 'firebase init' and set up your Storage configuration
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
````

**Reply by mikehardy** [reply]
That's the port. You're missing the rules. Examine our CI setup in the directly link above - firebase.json and storage.rules

**Reply by ithustle** [reply]
Oops. My bad. I'll try it. Thanks
